cmake/dep.cmake

```diff
@@ -13,6 +13,27 @@
 # limitations under the License.
 
 
+### Eigen ###
+
+set(EIGEN_FIND_REQUIRED true)
```

Comment:
I think we need to check the version of Eigen in CMake. Could you put a `TODO` here?